### PR TITLE
make max_connections configurable

### PR DESCRIPTION
**Pull request by mantel**

### CODE REVIEW

File: src/smdba/postgresqlgate.py [hunk 1]
```diff
@@ -903,14 +903,22 @@ def do_system_check(self, *args, **params):
...
         changed = False
+        max_connections = 400
 
         #
         # Setup postgresql.conf
         #
 
         # Built-in tuner
+        if params.get('max_connections') != None:
+            max_connections = int (params.get('max_connections'))
```
Comment (isbm) [inline src/smdba/postgresqlgate.py]: Here you could just do:
```python
max_conn = int(params.get('max_conn', 400))
```

File: src/smdba/postgresqlgate.py [hunk 2]
```diff
@@ -903,14 +903,22 @@ def do_system_check(self, *args, **params):
...
+
+        if max_connections < 400:
+            print >> sys.stdout, "INFO: max_connections should be at least 400."
+            max_connections = 400
+
         if 'autotuning' in args:
-            for item, value in PgTune().estimate().config.items():
+            for item, value in PgTune(max_connections).estimate().config.items():
```
Comment (isbm) [inline src/smdba/postgresqlgate.py]: Actually, all the code above can be just one-liner:
```python
for item, value in PgTune(max(int(params.get('max_connections', 400)), 400)).estimate().config.items():
...
```
Old school... :wink:

Comment (mantel) [inline src/smdba/postgresqlgate.py]: But then the admin will not be informed if the value supplied was too low!?

Comment (isbm) [inline src/smdba/postgresqlgate.py]: In this case it should not really inform, but either a) Fail and suggest to reconfigure or b) automatically set it up for the admin.

Comment (isbm) [inline src/smdba/postgresqlgate.py]: Still, if you want  to notify, then:
```python
conn_lowest = 400
max_conn = max(int(params.get('max_connections', conn_lowest)), conn_lowest)
if max_conn < conn_lowest:
    # notify here
    # change the config here
    pass
for item, value in PgTune(max_conn).estimate().config.items():
    ...
```

Comment (mantel) [inline src/smdba/postgresqlgate.py]: This condition can never be true, since max_conn will be at least conn_lowest, which is never less than conn_lowest.

Comment (isbm) [inline src/smdba/postgresqlgate.py]: But this is what you wanted in ``` if max_connections < 400:```, no?

Comment (mantel) [inline src/smdba/postgresqlgate.py]: No. If the admin passes a parameter that is smaller than 400, it should print a warning and set the parameter to the minimum of 400.

Comment (isbm) [inline src/smdba/postgresqlgate.py]: OK, I would propose it this way:
```python
conn_lowest = 400
max_conn = int(params.get('max_connections', conn_lowest))
if max_conn < conn_lowest:
    print >> sys.stdout, 'INFO: max_connections should be at least {0}'.format(conn_lowest)
    max_conn = conn_lowest
...
   for item, value in PgTune(max_conn).....
```

**Comment by isbm** [comment]
@mantel :tada:

**Pull request merged**